test/unit/block_test.rb

```diff
@@ -82,4 +82,26 @@ def test_exception_renderer_exception
     end
     assert_equal(handler_error, exc)
   end
+
+  def test_liquid_tag
```

Comment:
Should this test be in liquid, and not liquid-c?

Reply:
It's a base case for the following test, which uses `disable_liquid_c_nodes: true`, so it's nice to have here right beside it.

Reply:
There are already integration tests in the liquid repo that cover the same functionality.  This unit test just makes it clearer what is broken if the `test_liquid_tag_with_disable_liquid_c_nodes` test fails.

Actually, what would give us more confidence in `disable_liquid_c_nodes: true` would be to allow that to be set the default value for that option globally and to run all the integration tests with it defaulting to that value.  I tried doing that and it results in 5 LiquidTagTest test failures that this PR fixes.  I've just pushed a commit that does this instead of adding unit tests here.

Reply:
I didn't see a test case for `{% liquid ... assign`, but looking closer, other existing tests seem to cover :+1: